src/amo/components/FeaturedAddons/index.js

```diff
@@ -1,32 +1,54 @@
+/* eslint-disable react/no-unused-prop-types */
 import React from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { compose } from 'redux';
 
+import { getFeatured } from 'amo/actions/featured';
+import { setViewContext } from 'amo/actions/viewContext';
 import SearchResults from 'amo/components/SearchResults';
-import { loadFeaturedAddons } from 'amo/utils';
 import { ADDON_TYPE_EXTENSION, ADDON_TYPE_THEME } from 'core/constants';
+import { withErrorHandler } from 'core/errorHandler';
 import translate from 'core/i18n/translate';
-import { safeAsyncConnect } from 'core/utils';
+import { apiAddonType } from 'core/utils';
 
-import './FeaturedAddons.scss';
+import './styles.scss';
 
 
 export class FeaturedAddonsBase extends React.Component {
   static propTypes = {
     addonType: PropTypes.string.isRequired,
-    hasSearchParams: PropTypes.bool.isRequired,
+    dispatch: PropTypes.func.isRequired,
+    errorHandler: PropTypes.object.isRequired,
+    hasSearchParams: PropTypes.bool,
     i18n: PropTypes.object.isRequired,
     loading: PropTypes.bool.isRequired,
+    params: PropTypes.object.isRequired,
     results: PropTypes.array,
   }
 
   static defaultProps = {
     hasSearchParams: true,
   }
 
+  componentWillMount() {
+    this.props.dispatch(setViewContext(this.requestedAddonType()));
```

Comment:
This is not a big deal but I'm so used to use doing `const { dispatch } = this.props;` that seeing `this.props.dispatch` really threw me off. I know it's an extra line or two do make the `const` declaration but for the particular task of getting component props it's so common we do `const { prop } = this.props;` that I'd prefer we always did it so I can tell what's a prop.

That said I did a quick search in the project and `this.props.foo` happens about a dozen times, so it's fine to merge, I guess I just never noticed it or saw components using it. 🤷‍♀️